tools/wt_cmp_uri

```diff
@@ -0,0 +1,308 @@
+                    if abs(missing) >= 10:
+                        print(' total of {} missing entries here'.format(abs(missing)))
+                    print('EOF on {}'.format(cc2.name))
+                    print('{} records read'.format(nrecords))
+                    return
+                nrecords += 1
+                k2 = cc2.cursor.get_key()
+                v2 = cc2.cursor.get_value()
+                verbose('  key: {} value: {}'.format(show_kv(k2), show_kv(v2)))
+
+        if abs(missing) >= 10:
+            print(' total of {} missing entries here'.format(abs(missing)))
+            
+        # At this point, the keys are the same
+        if v1 != v2:
+            print('for key={}, value is different: {}: {}, {}: {}'.format(show_kv(k1), cc1.name, show_kv(v1), cc2.name, show_kv(v2)))
+        elif version:
+            # If we want to use the version cursor:
+            vcur1 = cc1.cursor.session.open_cursor(cc1.uri, None, 'debug=(dump_version)')
+            vcur1.set_key(k1)
+            if vcur1.search() != 0:
+                print('unexpected version cursor search')
+                return
+            vcur2 = cc2.cursor.session.open_cursor(cc2.uri, None, 'debug=(dump_version)')
+            vcur2.set_key(k2)
+            if vcur2.search() != 0:
+                print('unexpected version cursor search')
+                return
+            vcc1 = CompareCursor(vcur1, cc1.uri, 'version(key={}) for {}'.format(k1, cc1.name))
+            vcc2 = CompareCursor(vcur2, cc2.uri, 'version(key={}) for {}'.format(k2, cc2.name))
+            verbose('READY for version cursor')
+            compare_version_cursors(vcc1, vcc2)
+            vcur1.close()
+            vcur2.close()
+        verbose('{}: next'.format(cc1.name))
+
+    if cc2.cursor.next() == 0:
+        print('EOF on {}'.format(cc1.name))
+    print('{} records read'.format(nrecords))
+
+# Deeply compare two keys at all known timestamps using the version cursor.
+# Note: this function is not yet used.
+def compare_version_cursors(cc1, cc2):
+    nrecords = 0
+    ret1 = 0
+    ret2 = 0
+    while ret1 == 0:
+        if ret2 != 0:
+            print('EOF on {}'.format(cc2.name))
+            break
+        nrecords += 1
+        k1 = cc1.cursor.get_key()
+        v1 = cc1.cursor.get_value()
+        k2 = cc2.cursor.get_key()
+        v2 = cc2.cursor.get_value()
+        verbose('Version: GOT: {}:{} {}:{}'.format(k1,v1,k2,v2))
+        while k1 != k2:
+            if k1 < k2:
+                print('missing key={} on {}'.format(str(k1), cc2.name))
+                if cc1.cursor.next() != 0:
+                    print('EOF on {}'.format(cc1.name))
+                nrecords += 1
+                k1 = cc1.cursor.get_key()
+                v1 = cc1.cursor.get_value()
+            else:
+                print('missing key={} on {}'.format(str(k2), cc1.name))
+                if cc2.cursor.next() != 0:
+                    print('EOF on {}'.format(cc2.name))
```

Comment:
Same issue as above; we should not follow through with getting the key/value pair if the `next()` failed.

Reply:
Yes, I'll change to return here also.